_draft/2017-07-31-why-jwt.md

```diff
@@ -0,0 +1,52 @@
+# Authentication Exposure
+
+However sensenet ECM has been supporting authentication through NTLM and Basic for a long time, neither of them is a good choice for some scenarios. NTLM can be used only, if the authenticating user is a Windows domain user as well, Basic authentication on the other hand requires to send your user's credentials with every request. 
+We decided to implement a modern and easily usable and secure enough means to authenticate a user in the sensenet ECM environment.
+
+### Why JWT?
+There were a plethora of authentication methods in the wild, but we needed only one. 
+We were sure about a token based solution, so we just needed to examine them. 
+Token based solutions have the following advances in common:
++ stateless
++ transfers claims between two parties
++ scalable and decoupled
+  - scalable: you can extend the data stored in the payload
+  - decoupled: you have the choice to delegate the authentication tasks to a separate server
+
+As the list narrowed, only the following choices remained:
+Simple Web Tokens (SWT), Security Assertion Markup Language Tokens (SAML), JSON Web Token (JWT).
+
+Our pick became the JWT and that was not by chance. Here I collected some of the features, that made us determined about it:
+
++ industry standard ([RFC 7519](https://tools.ietf.org/html/rfc7519))
++ can be digitally signed and/or encrypted.
++ compact and self-contained
+  - compact:  JSON is less verbose than XML, so because of their small size, JWTs can be sent through the line as a POST parameter, or inside an HTTP header. The small size also means fast transmission.
+  - self-contained: the payload can contain every required information about the user and the scope, eliminating the need to query the database again an again.
++ works seamlessly with CORS and Single Sign On
++ JSON is easy to sign with a digital signature, however SAML tokens can use a public/private key pair in the form of a X.509 certificate it would make the signed token huge in size
++ JSON parser is already in JavaScript and it maps directly to objects. JS does not have document-to-object mapping for XML 
++ JWT is used at Internet widely, because the ease of JS processing of the JSON Web token on many platforms, particularly mobile.
+
+### Is That enough?
+Not really, because JWT is only a form for the information to get through. 
+It contains claims in its payload, but tells us nothing about the communication between our client and service.
+    ![Authentication Architecture](/img/authentication-architecture.png "Authentication Architecture")
+
+#### Architecture
+The illustration above depicts the logical architecture of the authentication. The phisical one however contains some simplification, so you do not need to set up a separate authentication server, it is an internal part of the sensenet ECM service layer.
+#### Signing
+We need to sign the JWT to make it easy to prove its thrustworthiness. A signed JWT is known as a JWS (JSON Web Signature). 
```

Comment:
The first sentence is a bit bumpy for me. What do we want to say? Can the sentence be easier?